libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/utils/InternalUtils.kt

```diff
@@ -1,27 +1,44 @@
 package com.mapbox.navigation.core.internal.utils
 
-import com.mapbox.navigation.core.trip.session.MapboxTripSession
+object InternalUtils {
 
-/**
- * Internal API used for testing. Sets the static unconditional polling patience value for all
- * instances in the process.
- *
- * Pass `null` to reset to default.
- *
- * Do not use in a production environment.
- */
-fun setUnconditionalPollingPatience(patience: Long?) {
-    MapboxTripSession.UNCONDITIONAL_STATUS_POLLING_PATIENCE = patience ?: 2000L
-}
+    internal var LOOK_AHEAD_MILLISECONDS = 1000L
+    internal var UNCONDITIONAL_POLLING_PATIENCE_MILLISECONDS = 2000L
+    internal var UNCONDITIONAL_POLLING_INTERVAL_MILLISECONDS = 1000L
+
+    /**
+     * Internal API used for testing. Sets the static look ahead value for all
+     * instances in the process. Needs to be set prior to [MapboxNavigation] creation.
+     *
+     * Pass `null` to reset to default.
+     *
+     * Do not use in a production environment.
+     */
+    fun setLookAhead(lookAheadInMillis: Long?) {
+        LOOK_AHEAD_MILLISECONDS = lookAheadInMillis ?: 1000L
+    }
```

Comment:
This parameter is important in scenarios where location is coming at a higher frequency than 1Hz. We should make it a part of the public API as it was before.